packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,137 @@ export interface AstroUserConfig {
 		 * For a complete overview and the full API reference, see [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
 		 */
 		contentLayer?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables a streamlined way of configuring TypeScript in your project for better DX when setting up Astro for Intellisense, type-checking, and more.
+		 *
+		 * This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding `outDir` by default.
+		 *
+		 * To enable this option in an existing project, you will need to update your existing `tsconfig.json` to reference the new file, and also potentially your `astro.config.mjs` as some items have now moved to Astro configuration options.
+		 *
+		 * Your `tsconfig.json` file must now extend at least the new `/.astro/` file:
+		 *
+		 * ```json title="tsconfig.json" ins="'./.astro/tsconfig.json'"
+		 * {
+		 * 	 extends: ['astro/tsconfigs/base', './.astro/tsconfig.json']
+		 * }
+		 * ```
+		 *
+		 * If you have `include` or `exclude` inside your root `tsconfig.json`, move them to your Astro config:
+		 *
+		 * ```json title="tsconfig.json" del={2-3}
+		 * {
+		 *   include: ['foo'],
+		 *   exclude: ['bar']
+		 * }
+		 * ```
+		 *
+		 * ```js title="astro.config.*" ins={6-7}
+		 * import { defineConfig } from 'astro/config'
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
+		 *    tsconfig: {
+		 *      include: ['foo'],
+		 *      exclude: ['bar']
+		 *    }
+		 *   }
+		 * })
+		 * ```
+		 */
+		tsconfig?: {
+			/**
+			 * @docs
+			 * @name experimental.tsconfig.include
+			 * kind: h4
+			 * @type {Array<string>}
+			 * @default `undefined`
+			 * @version 4.15.0
+			 * @description
+			 *
+			 * If you have `include` your root `tsconfig.json`, you'll need to move it to your Astro config:
+			 *
+			 * ```json title="tsconfig.json" del={2}
+			 * {
+			 *   include: ['foo']
+			 * }
+			 * ```
+			 *
+			 * ```js title="astro.config.*" ins={6}
+			 * import { defineConfig } from 'astro/config'
+			 *
+			 * export default defineConfig({
+			 *   experimental: {
+			 *    tsconfig: {
+			 *      include: ['foo']
+			 *    }
+			 *   }
+			 * })
+			 * ```
```

Comment:
``````suggestion
			 *
			 * Specifies an array of filenames or patterns to include in the program. These filenames are resolved relative to the directory containing the `tsconfig.json` file.
			 *
			 * ```js title="astro.config.*" {6}
			 * import { defineConfig } from 'astro/config'
			 *
			 * export default defineConfig({
			 *   experimental: {
			 *    tsconfig: {
			 *      include: ['src/**/*']
			 *    }
			 *   }
			 * })
			 * ```
``````
This is the API reference for the entry. We don't have to show upgrade guidance here, only what it is. We've shown how to convert above!